src/map.rs

```diff
@@ -1,38 +1,94 @@
-use std::{collections::HashMap, fmt, io::Read, path::Path, str::FromStr};
+use std::{
+    collections::HashMap,
+    fmt,
+    fs::File,
+    io::Read,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
 
-use xml::{attribute::OwnedAttribute, EventReader};
+use xml::{attribute::OwnedAttribute, reader::XmlEvent, EventReader};
 
 use crate::{
     error::{ParseTileError, TiledError},
     layers::{ImageLayer, Layer},
     objects::ObjectGroup,
-    properties::{parse_properties, Colour, Properties},
+    properties::{parse_properties, Color, Properties},
     tileset::Tileset,
-    util::*,
+    util::{get_attrs, parse_tag},
 };
 
 /// All Tiled files will be parsed into this. Holds all the layers and tilesets
 #[derive(Debug, PartialEq, Clone)]
 pub struct Map {
+    /// The TMX format version this map was saved to.
     pub version: String,
+    /// The orientation of this map.
```

Comment:
A bit of a nit (and very opinionated so worth someone else's opinion) but not sure comments like this add much

Reply:
I agree, I just have the tendency to always add docs to everything.